### ISSUE TITLE
rye is not specifying the correct VIRTUAL_ENV for uv.

### ISSUE THREAD

**Issue by jyomu** [astral-sh/rye#715]
### Steps to Reproduce

1. `rye config --set-bool behavior.use-uv=true`
2. `rye lock --verbose`

### Expected Result

generate lockfile.

### Actual Result

```
PS E:\AI\nerfstudio> rye config --set-bool behavior.use-uv=true
PS E:\AI\nerfstudio> rye lock --verbose
Python version already downloaded. Skipping.
Initializing new virtualenv in E:\AI\nerfstudio\.venv
Python version: cpython@3.10.13
 uv_interpreter::python_query::find_requested_python request="C:\\Users\\jyomu\\scoop\\persist\\rye\\py\\cpython@3.10.13\\install\\python.exe", platform=Platform { os: Windows, arch: X86_64 }, cache=Cache { root: "\\\\?\\C:\\Users\\jyomu\\AppData\\Local\\uv\\cache", refresh: None, _temp_dir_drop: None }
      0.002875s   0ms DEBUG uv_interpreter::interpreter Using cached markers for: \\?\C:\Users\jyomu\scoop\persist\rye\py\cpython@3.10.13\install\python.exe
Using Python 3.10.13 interpreter at C:\Users\jyomu\scoop\persist\rye\py\cpython@3.10.13\install\python.exe
Creating virtualenv at: E:\AI\nerfstudio\.venv
Generating production lockfile: E:\AI\nerfstudio\requirements.lock
 uv::requirements::from_source source=R:\tmp\.tmpgPeg4V
 uv::requirements::from_source source=R:\tmp\.tmp1z42AJ\requirements.txt
warning: Requirements file R:\tmp\.tmp1z42AJ\requirements.txt does not contain any dependencies
    0.004668s DEBUG uv_interpreter::virtual_env Found a virtualenv through VIRTUAL_ENV at: E:\AI\nerfstudio\venv
error: failed to canonicalize path `E:\AI\nerfstudio\venv\python.exe`
  Caused by: 指定されたパスが見つかりません。 (os error 3)
error: could not write production lockfile for project

Caused by:
    failed to generate lockfile
PS E:\AI\nerfstudio> ls

    Directory: E:\AI\nerfstudio

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----          2024/02/16    19:30                .devcontainer
d----          2024/01/12    20:26                .github
da---          2024/02/20    21:51                .venv
d----          2024/02/04     9:58                .vscode
d----          2024/02/04     9:58                colab
d----          2024/01/14    13:46                data
d----          2024/02/13    11:00                docs
d----          2024/02/03    22:08                exports
d----          2024/02/05    15:13                nerfstudio
d----          2024/02/20    21:45                nerfstudio.egg-info
d----          2024/02/15    21:01                outputs
d----          2024/02/04     9:58                tests
-a---          2024/02/04     9:58           3390 .dockerignore
-a---          2024/01/12    20:26             30 .gitattributes
-a---          2024/02/04     9:58           3391 .gitignore
-a---          2024/02/04     9:58            773 .pre-commit-config.yaml
-a---          2024/02/04     9:58            126 .prettierrc.js
-a---          2024/02/16    21:17              8 .python-version
-a---          2024/02/16    19:30           5943 Dockerfile
-a---          2024/02/04     9:58          11349 LICENSE
-a---          2024/01/12     9:54      196789599 PXL_20240107_004824524.TS.mp4
-a---          2024/02/13    14:40      101838738 PXL_20240213_053929315.TS.mp4
-a---          2024/02/16    20:55           5836 pyproject.toml
-a---          2024/02/12    22:28          21558 README.md

PS E:\AI\nerfstudio> $env:VIRTUAL_ENV
E:\AI\nerfstudio\venv
```

### Version Info

```
PS E:\AI\nerfstudio> rye --version
rye 0.25.0
commit: 0.25.0 (d8e00cea1 2024-02-19)
platform: windows (x86_64)
self-python: cpython@3.12
symlink support: true
uv enabled: true
```

**Comment by jyomu** [comment]
This seems to have occurred because the shell already had an environment variable. Sorry.

**Comment by mitsuhiko** [comment]
@jyomu That still sounds like a bug worth fixing. Are you saying it breaks if `VIRTUAL_ENV` is exported already?

**Comment by jyomu** [comment]
Yes.
Works correctly when `.venv` is specified, but error occurs when `venv` is specified
```
PS E:\AI\nerfstudio> $env:VIRTUAL_ENV=".venv"
PS E:\AI\nerfstudio> rye sync
Reusing already existing virtualenv
Generating production lockfile: E:\AI\nerfstudio\requirements.lock
   Built file:///E:/AI/nerfstudio                                                                                                   Built 1 editable in 3.68s
Resolved 238 packages in 4.76s
Generating dev lockfile: E:\AI\nerfstudio\requirements-dev.lock
   Built file:///E:/AI/nerfstudio                                                                                                   Built 1 editable in 2.19s
Resolved 238 packages in 2.27s
Installing dependencies
Audited 238 packages in 27ms
Done!
PS E:\AI\nerfstudio> $env:VIRTUAL_ENV="venv" 
PS E:\AI\nerfstudio> rye sync
Reusing already existing virtualenv
Generating production lockfile: E:\AI\nerfstudio\requirements.lock
error: failed to canonicalize path `venv\python.exe`
  Caused by: 指定されたパスが見つかりません。 (os error 3)
error: could not write production lockfile for project

Caused by:
    failed to generate lockfile
```
